### ISSUE TITLE
Reading Frame Buffer repeatedly

### ISSUE THREAD

**Issue by ImportanceOfBeingErnest** [vascotenner/lantz_drivers_xenics#1]
I want to continuously read the frame buffer of a Xenics camera using ctypes in python and the XCamera.dll from Xenics. 
Although I do not directly use the lantz_drivers_xenics, I actually copied much of the relevant code from it, so the issue may relate to the lantz_driver_xenics as well. 
Problem description: When reading out the frame buffer using the XC_CopyFrameBuffer() method, reads take up to half a second. I described the problem in more detail on stackoverflow 
[http://stackoverflow.com/questions/40294238/reading-frame-buffer-from-camera-using-ctypes](http://stackoverflow.com/questions/40294238/reading-frame-buffer-from-camera-using-ctypes)
The question is, is it normal to take so much time to read the frame? Is there something in the way the driver works that prevents reading at higher speed? Is this an issue with the lantz_driver or rather with the code I made from it?

(I cannot install lantz because I'm on python 2,.7 and everything on that machine relies on it)

**Comment by vascotenner** [comment]
I do not see this problem with python2.7 or 3.5 with lantz or with your code. The waiting time depends between 0 and 0.03s. Note that I never tried to use the highest possible framerates yet. Furthermore, the current timing is based on some try and error.  The Xcamera driver assumes that you do the timing yourself.

**Comment by ImportanceOfBeingErnest** [comment]
Thanks for the answer. If you did try my code and it did work on python 2.7 then you actually did try to use the highest possible framerate - because that is, what the code is doing, correct?
In this case, one would need to find out what else is different. 
I suppose that we do not have the exact same camera then? I'm using a Xenics Xeva 640. Could it be that it needs a different frame reading algorithm? Any idea how to find that out?

**Comment by vascotenner** [comment]
I have a 642. Some idea: 
- I have had some problems with the USB connection before.
- Does the camera operate fine with the xcontrol software? That uses the same dll under the hood.

**Comment by ImportanceOfBeingErnest** [comment]
Yes, the camera runs fine with the XControl Software and produces 24 frames per second. (Btw: Do you know how the xcontrol software reads out the framebuffer?)
I therefore concluded that the problem is not the camera itself or the USB connection or any dll-driver issue.

**Comment by vascotenner** [comment]
Very strange indeed. Btw, I tried this code while running windows XP.

**Comment by vascotenner** [comment]
There are some examples together with the xcamera SDK, that might show how the driver can be used. I assume that xcontrol uses the xcamera.dll directly, with comparable code.
